### ISSUE TITLE
Failed to refmt

### ISSUE THREAD

**Issue by razagill** [jaredly/reason-language-server#217]
Hi, I'm having a problem regarding the language server. I have a lerna monorepo setup where the `bsconfig.json` is inside a sub project and the dependencies are hoisted at the parent level.

This is the error from `node_modules/.lsp/debug.log`
```
{"jsonrpc":"2.0","id":4,"method":"textDocument/codeAction","params":{"textDocument":{"uri":"file:///Users/razagill/Development/nugen/shared/src/util/general.re"},"range":{"start":{"line":3,"character":0},"end":{"line":3,"character":1}},"context":{"diagnostics":[]}}}
[server] Got a method textDocument/codeAction
[server] processing took 0.0321865081787ms
Sending response {"id": 4, "jsonrpc": "2.0", "error": {"code": -32603, "message": "Failed to refmt '/Users/razagill/Development/nugen/node_modules/bs-platform/lib/refmt.exe' --print binary --recoverable --parse re > '/Users/razagill/Development/nugen/shared/node_modules/.lsp/General-@nugen/shared.ast'\n/bin/sh: /Users/razagill/Development/nugen/shared/node_modules/.lsp/General-@nugen/shared.ast: No such file or directory"}}
```
`version: 1.4.1`
`os: osx mojave 10.14.2`
`node: 11.0.0`
`bs-platform: 4.0.14`

**Comment by razagill** [comment]
@jaredly Absolutely. Although, I have no idea of how the language-server is working but I can help anyway I can.

**Comment by jaredly** [comment]
@razagill could you make a pull-request that adds a simple lerna monorepo project to the `examples/` directory, so I can test this out?

**Comment by nireno** [comment]
I see a similar message in my project but I'm not working with lerna (so let me know if to open a new issue). I just have a `bsconfig.json` with the name `@nireno/reasonably-typed`. This causes a `.lsp/<MyModuleName>-@nireno/reasonablyTyped.ast: Directory nonexistant` error to show at the top of the file.

If I remove the `/` from the name in bsconfig the error goes away.

What's strange is that everything still seems to work. The code gets formatted properly on save and the project builds.

Side note: I had to spread my vscode across two screens to read that error :-)

**Comment by MoOx** [comment]
Just got the same problem. It's not related to monorepo directly imo. As soon as I remove any / from the package name (in my case: `@phenomic/*`) it's ok. As soon as I add those back, I get an error on the path...

Eg: 
```
'/Users/moox/Sync/Development/phenomic/node_modules/bs-platform/lib/refmt.exe' --print binary --recoverable --parse re > '/Users/moox/Sync/Development/phenomic/packages/data-server-wordpress/node_modules/.lsp/WordpressApiV2Posts-@phenomic/dataServerWordpress.ast'
```

FYI, currently ocaml?bucklescript? (I mean without this reason-language-server - by hand) handle those namespace as Just camel case (eg: `@ns/something` became `NsSomething`, so file like `Else.re` will ends up as `Else-NsSomething.ast` instead of `Else-@ns/something.ast`

**Comment by MoOx** [comment]
To be clear what is triggering the issue is the conjunction of using [npm scope](https://docs.npmjs.com/misc/scope) in the name + [bucklescript `namespace`](
https://bucklescript.github.io/docs/en/build-configuration#name-namespace) feature.
Very easy to reproduce.

**Comment by MoOx** [comment]
workaround
- `package.json/name`: keep `@ns/*` (this one is used for npm)
- `bsconfig.json/name`: `ns-*` (this one will end up with the same as `@ns/` with normal reason tooling

**Comment by MoOx** [comment]
In addition to refmt not working, if you have a package this way, linked (via npm or yarn) and used in another project, `reason-language-server` will crash 5 times & stop. I guess it's related, cause as soon as I used the workaround in my project that created linked package in the consumer project, everything started to work around (without any crash).

**Comment by tizoc** [comment]
Mine is a regular repo, and I'm getting this error often. I just created `node_modules/.lsp` manually and the error went away (for now at least).
